stablehlo/dialect/StablehloBytecode.h

```diff
@@ -0,0 +1,28 @@
+/* Copyright 2022 The StableHLO Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_DIALECT_STABLEHLO_BYTECODE_H
+#define STABLEHLO_DIALECT_STABLEHLO_BYTECODE_H
+
+namespace mlir {
+namespace stablehlo {
+class StablehloDialect;
+
+// Add the interfaces necessary for encoding the stablehlo dialect components
+// in bytecode.
+void addBytecodeInterface(StablehloDialect *dialect);
```

Comment:
Based on BuiltinDialect: https://github.com/llvm/llvm-project/blob/df4e637ca7ef4ef17b662845120864921e65bb67/mlir/lib/IR/BuiltinDialectBytecode.h

Minimal risk of this going wrong, but I suppose using a more precise type prevents the accidental attempt to add the StableHLO bytecode interface to the CHLO dialect. Would be odd to attempt, but maybe with some bad copy-pasting it could happen.